test/e2e_test_policy_crd.sh

```diff
@@ -0,0 +1,270 @@
+#!/usr/bin/env bash
+#
+# Copyright 2022 The Sigstore Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -ex
+echo '::group:: invalid policy: both glob and regex'
+
+cat > policy.yaml <<EOF
```

Comment:
I wonder if we might want to create something like:
./test/testadata/success
./test/testdata/failure

directories and then create files there with names like:
./test/testdata/failure/both-glob-and-regex.yaml
./test/testdata/failure/key-with-multiple-properties.yaml

And then the both-glob-and-regex.yaml would look like:
```
# Invalid policy: both glob and regex
apiVersion: cosigned.sigstore.dev/v1alpha1
kind: ClusterImagePolicy
metadata:
  name: image-policy
spec:
  images:
  - glob: image**
    regex: image.*
    authorities:
    - key:
        data: "---somedata---"
```

Then we could have a for loop for both directories and for failures they should fail, and for successes they
should be created. For success case, then we could also delete after created, so that we don't have to deal
with conflicting names.

Just thinking that having those files might make it easier then to add new tests in the future as well as have some example configurations that could also serve as documentation. They could also be used for testing the validations for the fields. Just a thought.

Reply:
If we go with the proposed filenames under testdata, I'd suggest to add a `cosigned` prefix to know where they are been used.

Reply:
That makes sense. I will push them to new files.